FSA-547 add limited hooks

webecks opened the pull request:
[FSA-547](https://humanmade.atlassian.net/browse/FSA-547) - Separate videos from current sitemap into its own sitemap

This is related to https://github.com/humanmade/fox-sports-asia/pull/995

I only need to add a parameter to the original code to support multi post type.
**For example in msm-sitemap.php:**
- `Metro_Sitemap::date_range_has_posts( $date_stamp, $date_stamp, $post_type )` - (line 154) to be called in wp-cli.php
- `Metro_Sitemap::generate_sitemap_for_date( $date_stamp, $post_type )` - (line 156) to be called in wp-cli.php
- changed from `date_range_has_posts( $start_date, $end_date )` (line 384) -> `date_range_has_posts( $start_date, $end_date, $post_type = '' )`.
- changed from `$post_types_in = self::get_supported_post_types_in()` (line 390) -> `$post_types_in = empty( $post_type ) ? self::get_supported_post_types_in() : $wpdb->prepare( '%s', $post_type )`
- and more like this, by adding a parameter $post_type

I could not find a way to add hooks and filters around this.
If you have other ideas, please let me know. Thanks.

**There are 2 sitemaps generated from this code adjustment:**
- sitemap.xml
- sitemap-videos.xml

*I have:*
- [x] Written doc blocks and inline documentation on all new changes
- [x] Added the Jira reference as a prefix to the title of the PR
- [ ] Notes on how-to-test supplied in issue description on Jira
- [ ] WP CLI commands supplied in issue description on Jira
- [ ] As a developer, I've tested this on www-development.foxsportsasia.com

webecks committed:
FSA-547 add limited hooks

tareiking commented:
@webecks - was this meant to be opened against the `fs-master` branch? If so, can you edit the PR and change the base?

webecks committed:
Merge branch 'fs-master' into FSA-547-split-sitemap-by-posttype

webecks commented:
@tareiking I changed the base.
There were 2 conflicts between fs-master and branch FSA-547-split-sitemap-by-posttype.
I fixed and merged it. But just realized that it might need to be merged by someone else (perhaps you) after the code changes are approved.
Could you please check? Thanks.

tareiking commented on msm-sitemap.php:
```diff
@@ -89,6 +94,8 @@ public static function sitemap_rewrite_init() {
 			add_rewrite_rule( '^sitemap(\-.*)?.xml$', 'index.php?sitemap=true&type=$matches[1]', 'top' );
 		}
+
+		do_action( 'msm_sitemap_rewrite_rule' );
```
Need a docblock to explain this action here.

tareiking reviewed (changes requested):
@webecks - you'll need to add the docblock. Otherwise, looks okay for usage on Fox ATM.

webecks committed:
FSA-547 add docblock to msm_sitemap_rewrite_rule hook

webecks reviewed (commented):
@tareiking is the current docblock okay?

tareiking commented on msm-sitemap.php:
```diff
@@ -699,8 +715,8 @@ public static function build_root_sitemap_xml( $year = false ) {
 		$sitemap_time = strtotime( $sitemap_date );
-		$post_type = get_query_var( 'type' );
-		$post_type = empty( $post_type ) ? '' : $post_type;
+		$post_type = get_query_var( 'post_type' );
```
We should check if this post type is registered as well here using `post_type_exists`.

tareiking commented on msm-sitemap.php:
```diff
@@ -80,15 +85,26 @@ public static function sitemap_init() {
+		 *
+		 * @since 1.3.0
+		 *
+		 * @param type  $var Description.
```
This `param` isn't actually valid here.

tareiking reviewed (changes requested):
@webecks tested locally, this creates the sitemaps as expected. Added a few comments.. getting close!

webecks committed:
FSA-547 make changes based on comments

webecks commented:
@tareiking I have updated the file.

webecks committed:
FSA-547 another changes to post type check

tareiking reviewed (approved):
Approving recent changes, thanks @webecks
